components/runes/src/db/cache/index_cache.rs

```diff
@@ -160,8 +175,50 @@ impl IndexCache {
         _db_tx: &mut Transaction<'_>,
         ctx: &Context,
         etchings_counter: &mut u64,
+        bitcoin_tx: &bitcoin::Transaction,
+        inputs_counter: &mut u64,
     ) {
         let (rune_id, db_rune, entry) = self.tx_cache.apply_etching(etching, self.next_rune_number);
+
+        // Get the rune from the etching
+        let rune = etching.rune.unwrap_or_else(|| {
+            // If no rune name is provided, generate the rune's reserved name
+            Rune::reserved(
+                self.tx_cache.location.block_height,
+                self.tx_cache.location.tx_index,
+            )
+        });
+
+        // Only check if rune name is reserved when a rune name is explicitly provided
+        if let Some(provided_rune) = etching.rune {
+            if provided_rune.is_reserved() {
+                try_debug!(
+                    ctx,
+                    "Invalid rune for etching, reserved rune {}",
+                    provided_rune
+                );
+                return;
+            }
+        }
+
+        // Validate rune commitment
+        let is_valid_commitment = rune_etching_has_valid_commit_with_reconnect(
+            &mut self.bitcoin_client,
+            &self.bitcoin_client_config,
+            ctx,
+            bitcoin_tx,
+            &rune,
+            self.tx_cache.location.block_height as u32,
+            inputs_counter,
+        )
+        .await
+        .unwrap_or(false);
+
+        if !is_valid_commitment {
+            try_debug!(ctx, "Invalid rune commitment for etching {}", rune);
+            return; // Skip invalid etchings
```

Comment:
Wouldn't you need to undo the `self.tx_cache.apply_etching` here if this is the case?